src/bin/main.rs

```diff
@@ -179,10 +185,31 @@ fn main() {
         }
     };
 
-    let code_files = paths.code_files.as_ref().map(|files| {
-        PathUtil::list_all_files(&files[..])
-            .unwrap()
-            .into_iter()
+    if let Some(code_files) = &paths.code_files {
+        if let Some(code_paths) = &paths.code_paths {
+            if code_files.len() != code_paths.len() {
+                eprintln!(
+                    "If argument code_paths is given in the toml file, it must have as many elements as argument code_files",
+                );
+                return;
+            }
+        }
+    }
+    let code_files = paths.code_files.as_ref().map(|patterns| {
+        patterns
+            .iter()
+            .zip(
+                paths
+                    .code_paths
+                    .as_ref()
+                    .unwrap_or(&vec!["".to_string(); patterns.len()]),
```

Comment:
I wonder if `zip` is really helping here, especially as it implies to create a fallback vector even though the lengths have been checked. Maybe using `enumerate` and indexing directly into `code_paths` (using `[..]` as it cannot panic due to the length check if it is at least `Some(..)`) is simpler?

Reply:
Resolved through using your suggestion (at least the fallback vector problem)

```rust
let code_files = match (&path.code_files, &path.code_paths) {
  (Some(code_files), Some(code_paths)) => ...,
  (Some(code_files), None) => ...,
  (None, _) => Vec::new(),
};
```